<?xml version="1.0"?>
<!--
Version:  2017.07.31
Authors:  Members of the Team NAIST-Panasonic at the Amazon Robotics Challenge 2017
          Gustavo A. Garcia R. garcia-g at is.naist.jp (Captain), 
          Lotfi El Hafi, Felix von Drigalski, Wataru Yamazaki, Viktor Hoerig, Arnaud Delmotte, 
          Akishige Yuguchi, Marcus Gall, Chika Shiogama, Kenta Toyoshima, Pedro Uriguen, 
          Rodrigo Elizalde, Masaki Yamamoto, Yasunao Okazaki, Kazuo Inoue, Katsuhiko Asai, 
          Ryutaro Futakuchi, Seigo Okada, Yusuke Kato, and Pin-Chu Yang
*******************
Copyright 2017 Team NAIST-Panasonic 
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 
    http://www.apache.org/licenses/LICENSE-2.0 
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*******************
-->

<launch>
  <!-- Serial port for the optoforce sensor -->
  <arg name="port" default="/dev/ttyACM0"/>
  <!-- Update speed. Available values:
        "Stop"
        "1000Hz"
        "333Hz"
        "100Hz"
        "30Hz"
        "10Hz"
        -->
  <arg name="speed" default="100Hz"/>
  <!-- Internal filtering. Available values:
        "No"
        "500Hz"
        "150Hz"
        "50Hz"
        "15Hz"
        "5Hz"
        "1.5Hz"
        -->
  <arg name="filter" default="1.5Hz"/>
  <!-- Clear the sensors offset -->
  <arg name="zero" default="True"/>
  <!-- DAQ (Data Acquisition) Type. It can be single or multiple channel, and have 3 or 6 axis. Available values:
        "s-ch/3-axis"
        "m-ch/3-axis"	
        "s-ch/6-axis"
        -->
  <arg name="type" default="m-ch/3-axis"/>
  <!-- Numerical index of the first sensor. The rest (if this is a multi-channel DAC)
  will get the consecutive numbers -->
  <arg name="starting_index" default="0"/>
  <!-- Direction of scaling file -->
  <arg name="scaling_file" default="$(find tnp_optoforce)/config/to_g.yaml"/>
  <arg name="nodename" default="optoforce"/>    
  
  <node ns="$(arg nodename)"  name="$(arg nodename)" pkg="tnp_optoforce" type="optoforce_node.py" output="screen">
    <param name="port" value="$(arg port)"/>
    <param name="speed" value="$(arg speed)"/>
    <param name="filter" value="$(arg filter)"/>
    <param name="zero" value="$(arg zero)"/>
    <param name="type" value="$(arg type)"/>
    <param name="starting_index" value="$(arg starting_index)"/>
    <rosparam command="load" file="$(arg scaling_file)"/>
  </node>
</launch>
